lib/spack/spack/stage.py

```diff
@@ -210,7 +241,7 @@ def __init__(
             if self.name not in Stage.stage_locks:
                 sha1 = hashlib.sha1(self.name.encode('utf-8')).digest()
                 lock_id = prefix_bits(sha1, bit_length(sys.maxsize))
-                stage_lock_path = os.path.join(spack.paths.stage_path, '.lock')
+                stage_lock_path = os.path.join(self.path, '.lock')
```

Comment:
This has to be inside the stage root.  Otherwise every stage gets its own lockfile, and this will cause races when directories are created.  See the comment about all stages using one lock file.

Reply:
Good to know.  Thanks!